pkg/templates/nstemplatetiers/nstemplatetier_generator.go

```diff
@@ -143,20 +132,22 @@ func (g NSTemplateTierGenerator) GenerateAllManifests() (map[string][]byte, erro
 //       template: >
 //         <yaml-ns-template>
 // ------
-func (g NSTemplateTierGenerator) GenerateManifest(tier string) ([]byte, error) {
-	if nstemplatetierTmpl == nil {
-		return nil, errors.Wrapf(initTmplErr, "Cannot generate the '%s' NSTemplateTier manifest", tier)
-	}
+func (g NSTemplateTierGenerator) GenerateManifest(tier, namespace string) (toolchainv1alpha1.NSTemplateTier, error) {
 	revisions, ok := g.revisions[tier]
 	if !ok {
-		return nil, errors.Errorf("tier '%s' does not exist", tier)
+		return toolchainv1alpha1.NSTemplateTier{}, errors.Errorf("tier '%s' does not exist", tier)
 	}
-	type namespace struct {
-		Type     string
-		Revision string
-		Template string
+	obj := toolchainv1alpha1.NSTemplateTier{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: "toolchain.dev.openshift.com/v1alpha1",
```

Comment:
@alexeykazakov I needed to add it because the `expected` object which is generated from a YAML source has the info. Otherwise, the comparison would fail :/

Reply:
Hm... then maybe let's remove it from the test yamls too? :)
We are really not supposed to set group and kind in objets. As far as I understand it's up to the go client to map the go type (`toolchainv1alpha1.NSTemplateTier` in this case) to the registered GVK.
In best case this meta will be ignored (so it doesn't give us anything if we compare it in the tests). In worst case it can cause issues if we forget to update these hardcoded string when switching to a newer API version (so GVK will be different).

Reply:
what I can do, though, is compare the `ObjectMeta` and `Spec` portions of the `NSTemplateTier` objects, ignoring the `TypeMeta` (assuming it will be valid)

Reply:
> what I can do, though, is compare the ObjectMeta and Spec portions of the NSTemplateTier objects, ignoring the TypeMeta (assuming it will be valid): see de8f6e5

I prefer this option ^ since it doesn't require unnecessary code in production. So, it's isolated in the tests.

Reply:
ok, so let me revert 83a5965, then

Reply:
We still need to remove the entire `TypeMeta` :)

Reply:
🤦‍♂ (and done in 1460ce6)